src/sentry/snuba/discover.py

```diff
@@ -1640,6 +1639,79 @@ def find_histogram_params(num_buckets, min_value, max_value, multiplier):
     return HistogramParams(num_buckets, bucket_size, start_offset, multiplier)
 
 
+def find_span_histogram_min_max(span, min_value, max_value, user_query, params, data_filter=None):
```

Comment:
Should we be merging this with the original? or decompose the original into functions which they can both compose from? It feels like a lot of this code is almost identical minus the columns/span info and lack of data filter?

Reply:
yeah I've noticed it too. I can follow up with a clean up